test/e2e/cleanup/iam.go

```diff
@@ -0,0 +1,207 @@
+package cleanup
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
+	"github.com/go-logr/logr"
+
+	"github.com/aws/eks-hybrid/test/e2e/constants"
+	"github.com/aws/eks-hybrid/test/e2e/errors"
+)
+
+type IAMCleaner struct {
+	iamClient *iam.Client
+	logger    logr.Logger
+}
+
+func NewIAMCleaner(iamClient *iam.Client, logger logr.Logger) *IAMCleaner {
+	return &IAMCleaner{
+		iamClient: iamClient,
+		logger:    logger,
+	}
+}
+
+func (c *IAMCleaner) ListRoles(ctx context.Context, filterInput FilterInput) ([]string, error) {
+	var roles []string
+
+	paginator := iam.NewListRolesPaginator(c.iamClient, &iam.ListRolesInput{
+		PathPrefix: aws.String(constants.TestRolePathPrefix),
```

Comment:
don't we need to maintain the old cleanup for roles for a bit? at least for a few rounds until we are sure everything has been deleted?

Reply:
I'm going to keep the bash script around for a bit to cover this case.